src/components/sidepanel/StopTimetable.js

```diff
@@ -67,25 +76,46 @@ class StopTimetable extends Component {
       return hourVal;
     });
 
+    const {min, max} = timeRangeFilter;
+
     return (
       <TimetableGrid>
-        {byHourOrdered.map(([hour, times], idx) => (
-          <TimetableSection key={`hour_${hour}_${idx}`}>
-            <TimetableHour> {hour}</TimetableHour>
-            <TimetableTimes>
-              {times.map((departure, idx) => (
-                <TimetableDeparture
-                  selectedJourney={selectedJourney}
-                  key={`time_${idx}`}
-                  onClick={onSelectAsJourney}
-                  stop={stop}
-                  date={date}
-                  departure={departure}
-                />
-              ))}
-            </TimetableTimes>
-          </TimetableSection>
-        ))}
+        {byHourOrdered.map(([hour, times], idx) => {
+          let showTimes = times;
+
+          if (min !== "" || max !== "") {
+            const intHour = parseInt(hour.replace(":", ""), 10) / 100;
+            if (intHour < parseInt(min) || intHour > parseInt(max)) {
+              return null;
+            }
+          }
+
+          if (routeFilter) {
+            showTimes = times.filter((departure) =>
+              get(departure, "routeId", "").startsWith(routeFilter)
```

Comment:
Now you can filter only with routeId like "1006" or "1004 4". If we want to filter by route number, it should be like this:
```suggestion
              get(departure, "routeId", "").substring(1).replace(/^0+/, "").startsWith(routeFilter)
```